src/v0/destinations/klaviyo/transform.js

```diff
@@ -141,55 +175,67 @@ const trackRequestHandler = (message, category, destination) => {
           }
         });
       }
-      if (!payload.properties) {
+      if (!attributes.properties) {
         payload.properties = {};
       }
       if (itemArr.length > 0) {
-        payload.properties.items = itemArr;
+        attributes.properties.items = itemArr;
       }
     }
 
     // all extra props passed is incorporated inside properties
     let customProperties = {};
     customProperties = extractCustomFields(
       message,
-      customProperties,
+      attributes.customProperties,
       ['properties'],
       ecomExclusionKeys,
     );
     if (!isEmptyObject(customProperties)) {
-      payload.properties = {
-        ...payload.properties,
+      attributes.properties = {
+        ...attributes.properties,
         ...customProperties,
       };
     }
 
-    if (isEmptyObject(payload.properties)) {
+    if (isEmptyObject(attributes.properties)) {
       delete payload.properties;
     }
   } else {
-    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
-    payload.token = destination.Config.publicApiKey;
-    if (message.properties && message.properties.revenue) {
-      payload.properties.$value = message.properties.revenue;
-      delete payload.properties.revenue;
-    }
-    const customerProperties = createCustomerProperties(message);
-    if (destination.Config.enforceEmailAsPrimary) {
-      delete customerProperties.$id;
-      customerProperties._id = getFieldValueFromMessage(message, 'userId');
+    attributes = constructPayload(message, MAPPING_CONFIG[category.name]);
+    const value =
+      message.properties?.revenue || message.properties?.total || message.properties?.value;
+    if (value) {
+      attributes.value = value;
+      if (attributes.properties) {
+        if (attributes.properties.revenue) {
+          delete attributes.properties.revenue;
+        } else if (attributes.properties.total) {
+          delete attributes.properties.total;
+        } else if (attributes.properties.value) {
+          delete attributes.properties.value;
+        }
```

Comment:
Why these are separate else if? Can you explain the logic here?

Reply:
We are mapping any of `revenue || total || value` to `value` field. And deleting original one to avoid duplication.